qa/rpc-tests/test_framework/blocktools.py

```diff
@@ -44,30 +44,134 @@ def serialize_script_num(value):
         r[-1] |= 0x80
     return r
 
+
+def get_nBits(chainHeight):
+    # consensus.nPowAveragingWindow = 17
+    # genesis.nBits = 0x200f0f0f
+    # nBits stabilizes after 0x2d windows
+    nBits = 0x200f0f0f
+    r = min(chainHeight // 17, 45)
+    nBits -= r
+    #print "--- called with {} --> nBits = {}".format(chainHeight, nBits)
+    return nBits
```

Comment:
@alsala In official C++ node, nBits field is generated deterministically from last 17 blocks; it is constant in regtest only (since up/down adjustment is set to zero, so target is anchored to max one).
This algorithm is different from the one in official C++ node, so a block generated in this python client should be accepted by the official C++ client only in regtest, right?
If so, why not simplifying and hard-code max target only?

Reply:
As discussed offline, this is indeed an approximation (and a working one) for regtest target. Hard-coding max target would not work due to rounding behaviour of  `arith_uint256` type in division. Please comment this appropriately

Reply:
Done